Exiled.API/Features/Pickups/FirearmPickup.cs

```diff
@@ -0,0 +1,87 @@
+// -----------------------------------------------------------------------
+// <copyright file="FirearmPickup.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Features.Pickups
+{
+    using Exiled.API.Features.Items;
+
+    using InventorySystem.Items.Firearms;
+
+    using BaseFirearm = InventorySystem.Items.Firearms.FirearmPickup;
+
+    /// <summary>
+    /// A wrapper class for Firearm.
+    /// </summary>
+    public class FirearmPickup : Pickup
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FirearmPickup"/> class.
+        /// </summary>
+        /// <param name="itemBase">The base <see cref="BaseFirearm"/> class.</param>
+        public FirearmPickup(BaseFirearm itemBase)
+            : base(itemBase)
+        {
+            Base = itemBase;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="BaseFirearm"/> that this class is encapsulating.
+        /// </summary>
+        public new BaseFirearm Base { get; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether if the pickup are already distrubuted.
+        /// </summary>
+        public bool IsDistributed
+        {
+            get => Base.Distributed;
+            set => Base.Distributed = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="FirearmStatus"/>.
+        /// </summary>
+        public FirearmStatus Status
+        {
+            get => Base.NetworkStatus;
+            set => Base.NetworkStatus = value;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether how many ammo have this Firearm.
+        /// </summary>
+        public byte Ammo
+        {
+            get => Base.NetworkStatus.Ammo;
+            set => Base.NetworkStatus = new FirearmStatus(value, Base.NetworkStatus.Flags, Base.NetworkStatus.Attachments);
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether Flags have this Firearm.
```

Comment:
`Gets or sets the <see cref="FirearmStatusFlags">.`
like you write in FirearmStatus

Reply:
Done